src/encoded/static/components/experiment_series.js

```diff
@@ -85,85 +291,71 @@ class ExperimentSeriesComponent extends React.Component {
         super(props);
         this.state = {
             /** Audit objects keyed by dataset @id. Datasets w/o audits included with empty object */
-            auditsByDataset: {},
+            viewableDatasets: {},
         };
-        this.getDatasetAudits = this.getDatasetAudits.bind(this);
         this.getViewableDatasets = this.getViewableDatasets.bind(this);
     }
 
     componentDidMount() {
-        this.getDatasetAudits();
-    }
-
-    componentDidUpdate() {
-        const relatedDatasets = this.props.context.related_datasets;
-        if (relatedDatasets.length > 0) {
-            // If any @id in related_datasets isn't in this.state.auditsByDataset, retrieve all
-            // dataset audits. This can happen if the user edits the viewed ExperimentSeries
-            // object.
-            const datasetsInAudits = relatedDatasets.some(dataset => !!this.state.auditsByDataset[dataset['@id']]);
-            if (!datasetsInAudits) {
-                this.getDatasetAudits();
-            }
-        }
-    }
-
-    /**
-     * Retrieve the audits of all related datasets in the current ExperimentSeries object. Once
-     * all dataset audits have been retrieved, they're put into this.state.datasetAudit.
-     */
-    getDatasetAudits() {
-        const relatedDatasets = this.props.context.related_datasets;
-        if (relatedDatasets.length > 0) {
-            // Retrieve the audits of all related datasets.
-            const datasetAtIds = relatedDatasets.map(dataset => dataset['@id']);
-            Promise.all(datasetAtIds.map(datasetAtId => (
-                fetch(`${datasetAtId}?frame=audit`, {
-                    method: 'GET',
-                    headers: {
-                        Accept: 'application/json',
-                    },
-                }).then((response) => {
-                    // Convert each response response to JSON
-                    if (response.ok) {
-                        return response.json();
-                    }
-                    return Promise.resolve(null);
-                })
-            ))).then((datasetAudits) => {
-                // All dataset audits have been retrieved, or we got an error when trying to
-                // retrieve one or more of them.
-                const auditsByDataset = {};
-                if (datasetAudits && datasetAudits.length > 0) {
-                    datasetAudits.forEach((datasetAudit) => {
-                        auditsByDataset[datasetAudit['@id']] = datasetAudit.audit || {};
-                    });
-                }
-                this.setState({ auditsByDataset });
-            });
-        }
+        this.getViewableDatasets();
     }
```

Comment:
Is it a problem to not have `componentDidUpdate` here? If someone logs in while viewing the page, could the page become out of date?